src/time_val.rs

```diff
@@ -4,13 +4,63 @@
 
 use glib_sys;
 use std::mem;
+use std::fmt;
+use translate::*;
 
-pub use glib_sys::GTimeVal as TimeVal;
+pub struct TimeVal(glib_sys::GTimeVal);
 
 pub fn get_current_time() -> TimeVal {
     unsafe {
         let mut ret = mem::uninitialized();
         glib_sys::g_get_current_time(&mut ret);
-        ret
+        TimeVal(ret)
+    }
+}
+
+impl Default for TimeVal {
+    fn default() -> Self {
+        let ret = unsafe { mem::zeroed() };
+        TimeVal(ret)
+    }
+}
+
+impl fmt::Debug for TimeVal {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("TimeVal")
+            .field("tv_sec", &self.0.tv_sec)
+            .field("tv_nsec", &self.0.tv_usec)
```

Comment:
You should add setters/getters for these fields also, or otherwise it would be a bit useless

Reply:
ok, after a second thought, I drop the newtype for now.